desc/equilibrium/coords.py

```diff
@@ -107,9 +107,8 @@ def map_coordinates(  # noqa: C901
 
     profiles = get_profiles(inbasis + basis_derivs, eq)
 
-    # TODO: make this work for permutations of in/out basis
-    if outbasis == ("rho", "theta", "zeta"):
-        if inbasis == ("rho", "alpha", "zeta"):
+    if set(outbasis) == {"rho", "theta", "zeta"}:
+        if set(inbasis) == {"rho", "alpha", "zeta"}:
```

Comment:
The if guards here shouldn't be changed unless the coordinate mapping routines are also generalized to account for this ordering.

Right now this will give wrong results unless the ordering on the left also matches the ordering on the right because the arrays are assumed to be [radial, poloidal, toroidal] coordinates.

Reply:
Hmm, yeah that makes sense. I will revert it then. Should we at least show a message to the user then? Like, `Given inbasis {inbasis} and outbasis {outbasis} combination has a better implementation of map_coordinates. But the order of elements must be given in a specific way. Until this issue is resolved, consider rearranging your inputs.`. 
The old version of the `test_map_coordinates_derivative` was using a different ordering, was that deliberate then?

Reply:
the 3d root finding routine was general, it's the 1d ones in those if statements that are not

Reply:
Yes, the message I wrote is not an error but a suggestion to the user to use the better implementation. Anyway, someone can fix that `todo` when working on #1154 or some other related issue.